src/Stratis.FederatedPeg.Features.FederationGateway/Wallet/FederationWalletSyncManager.cs

```diff
@@ -79,6 +79,11 @@ public void Start()
                 ICollection<uint256> locators = this.walletManager.GetWallet().BlockLocator;
                 var blockLocator = new BlockLocator { Blocks = locators.ToList() };
                 ChainedHeader fork = this.chain.FindFork(blockLocator);
+                if (fork == null)
```

Comment:
this should absolutely never happen.

Reply:
```
        /// <summary>
        /// Returns the first chained block header that exists in the chain from the list of block hashes.
        /// </summary>
        /// <param name="hashes">Hash to search for.</param>
        /// <returns>First found chained block header or <c>null</c> if not found.</returns>
        public ChainedHeader FindFork(IEnumerable<uint256> hashes)
        {
            if (hashes == null)
                throw new ArgumentNullException("hashes");
            
            // Find the first block the caller has in the main chain.
            foreach (uint256 hash in hashes)
            {
                ChainedHeader mi = GetBlock(hash);
                if (mi != null)
                    return mi;
            }

            return null;
        }```

Reply:
![image](https://user-images.githubusercontent.com/29645989/50147260-c66afc80-0309-11e9-84bd-1323f5cca46d.png)